core/templates/dev/head/services/contextual/UrlService.js

```diff
@@ -54,6 +54,14 @@ oppia.factory('UrlService', ['$window', function($window) {
       }
       return topicId;
     },
+    getStoryIdFromUrl: function() {
+      var pathname = this.getPathname();
+      var storyId = pathname.split('/')[3];
```

Comment:
What if pathname.split('/') has fewer than 4 entries? Please be diligent in identifying the corner cases and making sure every possibility is accounted for.

Also ... tests? Please don't forget to write comprehensive unit tests; I shouldn't need to keep remarking on this in code reviews.

Reply:
Sorry, about that, missed it. Done.